aiida/Dockerfile

```diff
@@ -33,14 +33,14 @@ RUN /opt/conda/bin/pip install -r /tmp/requirements.txt && \
 # RENKU_VERSION determines the version of the renku CLI
 # that will be used in this image. To find the latest version,
 # visit https://pypi.org/project/renku/#history.
-ARG RENKU_VERSION=0.14.2
+ARG RENKU_VERSION=0.16.0
 
 ########################################################
 # Do not edit this section and do not add anything below
 
 RUN if [ -n "$RENKU_VERSION" ] ; then \
-    pipx uninstall renku && \
-    pipx install --force renku==${RENKU_VERSION} \
+    pip uninstall renku && \
```

Comment:
Simply using pip install here will install renku python together with all the other python packages which comes at a risk of colliding common dependencies. Avoiding this was the intention behind using pipx, which installed renku python into its own virtual environment. Replicating pipx without using pipx was the idea behind what I added https://github.com/SwissDataScienceCenter/renku-project-template/blob/4068f96df5512c082e75766289dad6972a577f3e/python-minimal/Dockerfile#L37. For this approach to work here you'd have to bump the base image to `>=0.10.1`.

Reply:
> comes at a risk of colliding common dependencies

For the moment the images built with the templates dont seem to have any colliding dependencies issue.

>For this approach to work here you'd have to bump the base image to >=0.10.1.

People that maintain and use these templates might not want/be able to work with python 3.9 yet and unfortuntately we dont have an image with a `3.8-0.10.1` tag available yet.

Reply:
> For the moment the images built with these templates dont seem to have any colliding dependencies issue.

I think the collisions are more likely to happen with packages that users will add to their "per-project" `requirements.txt`. I'd therefore try to stick to installing renku in a separate virtualenv. 

If upgrading the base image is not an option, you could create the virtualenv in this Dockerfile along the lines of what I did here: https://github.com/SwissDataScienceCenter/renkulab-docker/blob/4eed73105459025f45d4b854247e5b721b8c183b/docker/py/Dockerfile#L62

The other option of course is for us to stop messing with her image and to leave this decisions to @LiliGasser :)

Reply:
The base image can be upgraded to a Python 3.9 one. 

Can I leave that to the two of you?

Reply:
@LiliGasser I have updated the base image, could you to double check everything else is in order? and approve if so?
- streamlit https://renkulab.io/projects/pamela.delgado/test-streamlit-new-base-image
- aiida https://renkulab.io/projects/pamela.delgado/test-aiida-new-base-image